libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
+https://www.elastic.co/blog/beats-6-7-0-released[Beats 6.7.0 release blog].
+
+[[migrate-config-files]]
 ==== Migrate configuration files
 
-Beats 6.0 comes with several backwards incompatible configuration changes.
-Please review the <<breaking-changes-6.0>> document. Where possible, we kept the
-old configuration options working, but deprecated them.  However, deprecation
-was not always possible, so if you use any of the settings mentioned in the
-Breaking Changes section of the release notes, make sure you understand the
+{beats} 7.0 comes with several backwards incompatible configuration changes.
+Before upgrading, review the <<breaking-changes-7.0>> document. Also review
+the full list of breaking changes in the <<release-notes>> for 7.0.
+
+Where possible, we kept the old configuration options working, but deprecated
+them. However, deprecation was not always possible, so if you use any of the
+settings described under breaking changes, make sure you understand the
 alternatives that we provide.
 
-===== modules.d configuration layout
+[[enable-ecs-compatibility]]
+==== Enable the compatibility layer for Elastic Common Schema (ECS) fields
 
-Starting with the 6.0 version, Filebeat and Metricbeat are moving to a directory
-layout for configuration, where each module is configured in its own
-configuration file.
+Starting with 7.0, the fields exported by {beats} conform to the
+{ecs-ref}/index.html[Elastic Common Schema (ECS)]. Many of the exported fields
+have been renamed. See {beats-ref}/breaking-changes-7.0.html[Breaking
+changes in 7.0] for the full list of changed names.
 
-While the all-in-one configuration is still fully supported, we recommend moving
-to the new layout at upgrade time. This typically means starting off with the new
-default configuration and modifying it with the custom settings that you had in
-your old files.
+To help you transition to the new fields, we provide a compatibility layer in
+the form of ECS-compatible field aliases. To use the aliases, set
+`migration.6_to_7.enabled: true` in the Beat's configuration file *before* you
+run {beats} 7.0 to ingest data for the first time.
 
-// TODO: link to the docs on modules.d
+The compatibility layer enables you to use 6.x dashboards and visualizations
+with indices created by {beats} 7.0 or later.
 
-===== New command `test config` command
+Did you run the Beat before reading this section? That's OK. See the clean-up
+steps described under <<missing-fields>>.
 
-Beats 6.0 introduces a new test command for testing the configuration file. For
-example:
+[[upgrade-mapping-template]]
+==== Upgrade the {es} mapping template
 
-[source,shell]
+Mapping templates and the default index names are versioned. For example,
+Metricbeat {stack-version} typically creates indices like this:
+
+["source","sh",subs="attributes"]
 ------------------------------------------------------------------------------
-metricbeat test config
+metricbeat-{stack-version}-{localdate}
```

Comment:
Seems the output (at least in Github) of localdate is `2019-04-02` but the index names are `2019.04.02`.

Reply:
Whoops. I was confused about the format that was returned by localdate. I will hard code.